### PR TITLE
fix(harvard_opinions): Handle bad ibid citaitons

### LINKED ISSUE #2351: Harvard Import Error
This error is being thrown ... occasionally on imports.  Looks like a change in January is now raising its head.
```
Traceback (most recent call last):
  File "/opt/courtlistener/manage.py", line 15, in <module>
    main()
  File "/opt/courtlistener/manage.py", line 11, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/opt/courtlistener/cl/corpus_importer/management/commands/harvard_opinions.py", line 1141, in handle
    parse_harvard_opinions(options)
  File "/opt/courtlistener/cl/corpus_importer/management/commands/harvard_opinions.py", line 360, in parse_harvard_opinions
    previously_imported_case = find_previously_imported_cases(
  File "/opt/courtlistener/cl/corpus_importer/management/commands/harvard_opinions.py", line 889, in find_previously_imported_cases
    citations__reporter=found_cite[0].corrected_reporter(),
AttributeError: 'IdCitation' object has no attribute 'corrected_reporter'
Sentry is attempting to send 2 pending error messages
Waiting up to 2 seconds
Press Ctrl-C to quit
```

There is even a warning in pycharm `Unresolved attribute reference 'corrected_reporter' for class 'CitationBase' `

```
    possible_cases = OpinionCluster.objects.filter(
        citations__reporter=found_cite[0].corrected_reporter(),
        citations__volume=found_cite[0].groups["volume"],
        citations__page=found_cite[0].groups["page"],
    ).order_by("id")
```

### PR DESCRIPTION
**Pull request by flooie**
Fix #2351 

PR addresses handling of citations that are not full case citations.  

Currently the importer crashes when a citation is malformed in such 
a way that it returns something other than a full case citation.  

This PR checks citations now and adds a test to filter out such citations

### CODE REVIEW

**Comment by flooie** [comment]
@mlissner 👀  if you dont mind

File: cl/tests/providers.py [hunk 1]
```diff
@@ -91,8 +83,21 @@ def case_name(self, full: bool = False) -> str:
...
-        """Make a citation e.g. 345 Mass. 76"""
+        """Make or fetch a citation e.g. 345 Mass. 76
+
+        Grab a random reporter if it contains a typical full_cite pattern
+        Exclude reporters that rely on regex patterns.
+
+        :return Citation as a string
+        """
+
+        reporters = [
```
Comment (mlissner) [inline cl/tests/providers.py:94]:
```suggestion
        # Filter to only vol-reporter-page reporters
        reporters = [
```

**Comment by quevon24** [comment]
@flooie The tests look good, only a few lines are too long in factories.py and tests.py

**Comment by mlissner** [comment]
> @flooie The tests look good, only a few lines are too long in factories.py and tests.py

I just merged this, but if you want to do formatting tweaks in your next PR, that'd work well.

**Pull request merged**